### DISCUSSION TITLE
Allow rule customization while enforcing that a specific set of rules ran?

### DISCUSSION THREAD

**Discussion by bichuga** [dotnet/roslyn#56185] [Q&A]
I have some folks asking about enabling specific Roslyn analyzers in their project, but we currently have such customization blocked because we don't want to open the door to people turning off security analyzers. Is there a way to achieve the scenario of allowing rule customization while ensuring certain rules are set? Imagine a giant git repository with many teams where everyone wants different stylecop/maintainability rules.

**Comment by jmarolf** [comment]
> but we currently have such customization blocked

How do you block it? Its hard to suggest possible solutions without understanding your constraints

**Reply by bichuga** [reply]
We have a special homebrew build tool that looks for prohibited properties in the building project. No project is allowed to specify <CodeAnalysisRuleSet> or it will break the build.

**Reply by jmarolf** [reply]
What properties? MSBuild properties can control whether codeanalysis in run on build or not but not what set are run. What if someone disables rules via an editorconfig file?

**Reply by bichuga** [reply]
We disallow people to set "CodeAnalysisRuleSet", "TreatWarningsAsErrors" without a special sign off on the pull request. You're right this doesn't help with the editorconfig case... it's more of a "help people not accidentally change a config they weren't supposed to." This all comes back to the original question -- we want to ensure a specific set of analyzers ran while also providing the opportunity for people to add their own set on top. Any possibilities you can think of?
